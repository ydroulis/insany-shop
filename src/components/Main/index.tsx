import React from 'react'
import * as S from './styles'

const Main = ({
    title = 'teste',
    description = 'Testando'
}) => (
    <S.Wrapper>
        <S.Title>{title}</S.Title>
        <S.Description>{description}</S.Description>
    </S.Wrapper>
)

export default Main