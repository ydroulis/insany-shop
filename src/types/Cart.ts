
export type CartProduct = {
    id: number
    name: string
    description: string
    price: number
    image: string
    stock: number
    items?: number
}

export type Cart = {
    content: CartProduct[]
    total: number
}