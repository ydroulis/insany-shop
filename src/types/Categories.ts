export type Categories = Category[];

export type Category = {
    id: string,
    name: string,
    description: string,
    icon: string,
    productCount: number
};