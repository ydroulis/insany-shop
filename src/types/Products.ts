export type Products = Product[]

export type Product = {
    id: number,
    name: string,
    description: string,
    price: number,
    image: string,
    category: string,
    stock: number,
    rating: number,
    brand?: string

}