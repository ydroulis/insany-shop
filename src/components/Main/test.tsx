import React from "react";
import { render, screen, within } from "@testing-library/react";
import Main from "./";
import { Product } from "@/types/Products";

jest.mock("../Filters", () => {
    const mockFilters = () => <div data-testid="filters-mock">Filtros</div>;
    return mockFilters;
});

jest.mock("../ProductList", () => {
    const ProductList = ({ products }: { products: Product[] }) => (
        <ul data-testid="product-list-mock">
            {products.map((product) => (
                <li key={product.id}>
                    <img
                        src={product.image}
                        alt={`Imagem do produto ${product.name}`}
                    />
                    <span>{product.name}</span>
                    <span>{product.category}</span>
                    <span>R$ {product.price.toFixed(2)}</span>
                    <span>{product.stock} em estoque</span>
                    <span aria-label={`Avaliação ${product.rating}`}>
                        ⭐ {product.rating}
                    </span>
                    <span>{product.brand}</span>
                </li>
            ))}
        </ul>
    );

    return ProductList;
});

describe("<Main />", () => {
    beforeEach(() => {
        render(<Main />);
    });

    it("should render the main landmark with correct accessibility attributes", () => {
        const main = screen.getByRole("main", { name: /catálogo de produtos/i });
        expect(main).toBeInTheDocument();
        expect(main.tagName.toLowerCase()).toBe("main");
    });

    it("should render the Filters section", () => {
        expect(screen.getByTestId("filters-mock")).toBeInTheDocument();
        expect(screen.getByText("Filtros")).toBeVisible();
    });

    it("should render the ProductList with products", () => {
        const list = screen.getByTestId("product-list-mock");
        expect(list).toBeInTheDocument();

        const items = within(list).getAllByRole("listitem");
        expect(items.length).toBeGreaterThan(0);

        const firstItem = items[0];
        expect(within(firstItem).getByRole("img")).toHaveAttribute(
            "alt",
            expect.stringMatching(/imagem do produto/i)
        );
        expect(within(firstItem).getByText(/iPhone/i)).toBeInTheDocument();
        expect(within(firstItem).getByText(/Apple/i)).toBeInTheDocument();
    });

    it("should display price, stock and rating correctly", () => {
        const list = screen.getByTestId("product-list-mock");
        const items = within(list).getAllByRole("listitem");

        items.forEach((item) => {
            const price = within(item).getByText(/R\$/i);
            expect(price).toBeInTheDocument();

            const stock = within(item).getByText(/em estoque/i);
            expect(stock).toBeInTheDocument();

            const rating = within(item).getByLabelText(/avaliação/i);
            expect(rating).toBeInTheDocument();
        });
    });

    it("should ensure all product images have accessible alt text", () => {
        const images = screen.getAllByRole("img");
        images.forEach((img) => {
            expect(img).toHaveAttribute("alt");
            expect(img.getAttribute("alt")).toMatch(/imagem do produto/i);
        });
    });
});
