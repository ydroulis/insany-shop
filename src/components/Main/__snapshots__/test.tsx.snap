// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Main/> should render the heading 1`] = `
<main
  class="sc-bRKDuR cQIcRK"
>
  <h1
    class="sc-hvigdm ebRyxB"
  >
    teste
  </h1>
  <h2
    class="sc-fhHczv eEWtjj"
  >
    Testando
  </h2>
</main>
`;
